mds/apis/prv_api/provider_api.py

```diff
@@ -0,0 +1,99 @@
+from rest_framework import decorators, pagination, response, serializers, viewsets
+
+from mds import enums, models, utils
+from mds.access_control.permissions import require_scopes
+from mds.access_control.scopes import SCOPE_PRV_API
+from mds.apis import utils as apis_utils
+
+
+class DeviceStatusChangesSerializer(serializers.ModelSerializer):
+    associated_trip = serializers.CharField(source="properties.trip_id")
+    device_id = serializers.CharField(source="device.id")
+    event_location = serializers.SerializerMethodField()
+    event_time = apis_utils.UnixTimestampMilliseconds(source="timestamp")
+    event_type = serializers.SerializerMethodField()
+    event_type_reason = serializers.CharField(source="event_type")
+    propulsion_type = serializers.ListSerializer(
+        source="device.propulsion",
+        child=serializers.ChoiceField(choices=enums.choices(enums.DEVICE_PROPULSION)),
+    )
+    provider_id = serializers.CharField(source="device.provider.id")
+    provider_name = serializers.CharField(source="device.provider.name")
+    vehicle_id = serializers.CharField(source="device.identification_number")
+    vehicle_type = serializers.CharField(source="device.category")
+
+    class Meta:
+        model = models.EventRecord
+        fields = (
+            "associated_trip",
+            "device_id",
+            "event_location",
+            "event_time",
+            "event_type",
+            "event_type_reason",
+            "propulsion_type",
+            "provider_name",
+            "provider_id",
+            "vehicle_id",
+            "vehicle_type",
+        )
+
+    def create(self, data):
+        raise NotImplementedError()
+
+    def update(self, instance, data):
+        raise NotImplementedError()
+
+    def get_event_location(self, obj):
+        telemetry = obj.properties.get("telemetry", {})
+        return {
+            "type": "Feature",
+            "properties": {"timestamp": telemetry.get("timestamp")},
+            "geometry": obj.point_as_geojson,
```

Comment:
Did you notice the `json.loads`? It returns a serialized string for... REASONS! https://github.com/Polyconseil/django-mds/blob/master/mds/models.py#L215